docs/source/llms/langchain/autologging.rst

```diff
@@ -0,0 +1,262 @@
+
+Customize Callback
+^^^^^^^^^^^^^^^^^^
+
+Sometimes you may want to customize what information is logged in the traces. You can achieve this by creating a custom callback handler that inherits from `MlflowLangchainTracer <https://github.com/mlflow/mlflow/blob/master/mlflow/langchain/langchain_tracer.py>`_. The following example demonstrates how to record an additional attribute to the span when a chat model starts running.
+
+.. code-block::
+
+    from mlflow.langchain.langchain_tracer import MlflowLangchainTracer
+
+    class CustomLangchainTracer(MlflowLangchainTracer):
+
+        # Override the handler functions to customize the behavior. The method signature is defined by LangChain Callbacks.
+        def on_chat_model_start(
+            self,
+            serialized: Dict[str, Any],
+            messages: List[List[BaseMessage]],
+            *,
+            run_id: UUID,
+            tags: Optional[List[str]] = None,
+            parent_run_id: Optional[UUID] = None,
+            metadata: Optional[Dict[str, Any]] = None,
+            name: Optional[str] = None,
+            **kwargs: Any,
+        ):
+            """Run when a chat model starts running."""
+            attributes = {
+                **kwargs,
+                **metadata,
+                # Add additional attribute to the span
+                "version": "1.0.0",
+            }
+
+            # Call the _start_span method at the end of the handler function to start a new span.
+            self._start_span(
+                span_name=name or self._assign_span_name(serialized, "chat model"),
+                parent_run_id=parent_run_id,
+                span_type=SpanType.CHAT_MODEL,
+                run_id=run_id,
+                inputs=messages,
+                attributes=kwargs,
+            )
+
+Patch Functions for Logging Artifacts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Other artifacts such as models, datasets, are logged by patching the invocation functions of the supported models to insert the logging call. MLflow patches the following functions:
+
+* ``invoke``
+* ``batch``
+* ``stream``
+* ``get_relevant_documents`` (for retrievers)
+* ``__call__`` (for Chains and AgentExecutors)
+* ``ainvoke``
+* ``abatch``
+* ``astream``
+
+.. warning::
+
+    MLflow supports autologging for async functions (e.g., ``ainvoke``, ``abatch``, ``astream``), however, the logging operation is not
+    asynchronous and may block the main thread. The invocation function itself is still not blocking and returns a coroutine object, but
+    the logging overhead may slow down the model inference process. Please be aware of this side effect when using async functions with autologging.
+
+Troubleshooting
+---------------
+
+If you encounter any issues with LangChain autologging, please refer to `FAQ <../index.html#faq>` first. If you still have questions, please feel free to open an issue in `MLflow Github repo <https://github.com/mlflow/mlflow/issues>`_.
```

Comment:
Do you think it's useful to add a FAQ about how to disable warning? Some people might don't want to see the warnings at all when they use autolog 🤔

Reply:
Great idea!